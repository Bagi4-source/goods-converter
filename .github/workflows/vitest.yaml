name: 'Test'
on:
  pull_request:
  push:

jobs:
  vitest:
    name: Run vitest
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
      - uses: actions/checkout@v2
      - name: 'Install Node'
        uses: actions/setup-node@v2
        with:
          node-version: '18.x'
      - name: 'Install Deps'
        run: npm i
      - name: 'Test'
        run: npm run test
      - name: 'Coverage'
        run: npm run coverage
      - name: 'Report Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2