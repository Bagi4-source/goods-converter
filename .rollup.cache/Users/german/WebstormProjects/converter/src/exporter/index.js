export * from "./exporter";
export * from "./exporter.types";
//# sourceMappingURL=index.js.map