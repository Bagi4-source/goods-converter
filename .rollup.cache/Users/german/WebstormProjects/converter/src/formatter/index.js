export * from "./CSV.formatter";
export * from "./YML.formatter";
export * from "./formater.types";
//# sourceMappingURL=index.js.map