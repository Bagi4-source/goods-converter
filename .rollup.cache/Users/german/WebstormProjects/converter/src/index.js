export * from "./exporter";
export * from "./formatter";
//# sourceMappingURL=index.js.map