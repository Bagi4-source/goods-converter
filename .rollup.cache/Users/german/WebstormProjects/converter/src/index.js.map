{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,YAAY,CAAA;AAC1B,cAAc,aAAa,CAAA","sourcesContent":["export * from \"./exporter\"\nexport * from \"./formatter\""]}