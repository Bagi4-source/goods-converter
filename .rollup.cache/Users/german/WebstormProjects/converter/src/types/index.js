export * from "./Product.types";
export * from "./Category.types";
//# sourceMappingURL=index.js.map