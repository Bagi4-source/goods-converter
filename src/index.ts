export * from "./exporter"
export * from "./formatter"
export * from "./types"