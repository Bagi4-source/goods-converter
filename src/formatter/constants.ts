import {CSVFormatter} from "./CSV.formatter";
import {YMLFormatter} from "./YML.formatter";
import {TildaFormatter} from "./Tilda.formatter";

export const Formatters = {
    CSV: new CSVFormatter(),
    YML: new YMLFormatter(),
    Tilda: new TildaFormatter(),
}
