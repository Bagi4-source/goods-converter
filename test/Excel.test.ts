import { expect, describe, it } from "vitest";

import { Formatters } from "../src";
import { categories, products } from "./constants";
import { hashStream } from "./util";

describe.skip("Excel formatter", () => {
  const formatter = new Formatters.ExcelFormatter();

  it("should export Excel data", async () => {
    const result = await formatter.format(products, categories);
    expect(await hashStream(result)).toMatchSnapshot();
  });
});
