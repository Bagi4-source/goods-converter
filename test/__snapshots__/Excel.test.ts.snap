// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Excel formatter > should export Excel data 1`] = `"f9c04ad32df9fea42c3a810516956da9d553de3ba478bf5b0fb473e1b2c645fa"`;
